clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12910,6 +12910,18 @@ def err_sycl_special_type_num_init_method : Error<
   "types with 'sycl_special_class' attribute must have one and only one '__init' "
   "method defined">;
 
+// SYCL external attribute diagnostics
+def err_sycl_attribute_invalid_linkage : Error<
+  "'clang::sycl_external' can only be applied to functions with external"
+  " linkage">;
+def err_sycl_attribute_invalid_main : Error<
+  "'clang::sycl_external' cannot be applied to the 'main' function">;
+def err_sycl_attribute_invalid_deleted_function : Error<
+  "'clang::sycl_external' cannot be applied to an explicitly deleted function">;
+def warn_attribute_missing_on_first_decl : Warning<
+  "%0 attribute does not appear on the first declaration">,
+  InGroup<IgnoredAttributes>;
```

Comment:
I don't think we should ignore the attribute if it is not present on the first declaration since we can still honor its semantic effects. Assuming agreement, this warning shouldn't be a member of the `IgnoredAttributes` group.

Additionally, this diagnostic has either an odd name or an odd placement since it is colocated with `sycl_external` diagnostics, but is not named to be specific to them. I recommend giving it a `sycl_external` specific name to make it easier to search for.
```suggestion
def warn_sycl_external_missing_on_first_decl : Warning<
  "%0 attribute does not appear on the first declaration">;
```

Reply:
Did you mean to suggest removing a grouping altogether? 
It is my understanding that every warning without a -W flag must belong to a group, else, it needs to be added to clang/test/Misc/warning-flags.c; the test fails otherwise.